### PR TITLE
Cleanup of start_date and default arg use for Apache example DAGs

### PR DESCRIPTION
**Pull request by josh-fell**
There is an ongoing effort to enhance example DAGs by setting static values for `start_date` (already complete for core example DAGs), cleaning up and/or implementing useful `default_args`, and removing/limiting the use of redundant, default connection ID values.

This PR mainly addresses these 3 cleanup areas in example DAGs across Apache providers.  There are other small docs updates and use of the `TriggerRule` attr rather than the literal string where applicable.

> Note: In the `example_beam` DAG, there was an opportunity to consolidate task args to `default_args` for a couple of DAGs, however, this was not implemented as the changes would have conflicted too much with the operator documentation that references the example DAG.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/apache/drill/example_dags/example_drill_dag.py [hunk 1]
```diff
@@ -19,14 +19,15 @@
...
+
 from airflow.models import DAG
 from airflow.providers.apache.drill.operators.drill import DrillOperator
-from airflow.utils.dates import days_ago
 
 with DAG(
     dag_id='example_drill_dag',
     schedule_interval=None,
-    start_date=days_ago(2),
+    start_date=datetime(2021, 1, 1),
```
Comment (zkan) [inline airflow/providers/apache/drill/example_dags/example_drill_dag.py:30]: Should we use `START_DATE` here like https://github.com/apache/airflow/pull/18657/files#diff-80e53751469787f4571bdcf652c0803676ab5e19b91c1dab7a39c678e9e7bfe3R76, so it will be consistent for all example DAGs?

Comment (josh-fell) [reply airflow/providers/apache/drill/example_dags/example_drill_dag.py:30]: I chose to create a `START_DATE` variable in that Apache Beam example DAG file because there are several DAGs declared in that single file -- all of which had the same `start_date` value.  IMO it doesn't add much value to do the same here as there is only one DAG declared so `start_date` is set once.

File: airflow/providers/apache/druid/example_dags/example_druid_dag.py [hunk 2]
```diff
@@ -19,22 +19,19 @@
...
+    start_date=datetime(2021, 1, 1),
     tags=['example'],
 ) as dag:
     # [START howto_operator_druid_submit]
-    submit_job = DruidOperator(
-        task_id='spark_submit_job',
-        json_index_file='json_index.json',
-        druid_ingest_conn_id='druid_ingest_default',
-    )
+    submit_job = DruidOperator(task_id='spark_submit_job', json_index_file='json_index.json')
```
Comment (uranusjr) [inline airflow/providers/apache/druid/example_dags/example_druid_dag.py:35]: Why remove `druid_ingest_conn_id` here, because it's the default?

File: airflow/providers/apache/hive/example_dags/example_twitter_dag.py [hunk 3]
```diff
@@ -77,8 +76,9 @@ def transfer_to_db():
         'retries': 1,
     },
     schedule_interval="@daily",
-    start_date=days_ago(5),
+    start_date=datetime(2021, 1, 1),
     tags=['example'],
+    catchup=False,
```
Comment (uranusjr) [inline airflow/providers/apache/hive/example_dags/example_twitter_dag.py:81]: Why this change?

File: airflow/providers/apache/druid/example_dags/example_druid_dag.py (hunk 2 above)
Comment (josh-fell) [reply airflow/providers/apache/druid/example_dags/example_druid_dag.py:35]: Yes. We've been removing these default connection IDs across the example DAGs. Do you feel differently?

File: airflow/providers/apache/hive/example_dags/example_twitter_dag.py (hunk 3 above)
Comment (josh-fell) [reply airflow/providers/apache/hive/example_dags/example_twitter_dag.py:81]: Only controlling the number of DAG runs that spin when users first enable the DAG now that we're transitioning to a static `start_date` value as best practice.

File: airflow/providers/apache/druid/example_dags/example_druid_dag.py (hunk 2 above)
Comment (uranusjr) [reply airflow/providers/apache/druid/example_dags/example_druid_dag.py:35]: No, just making sure.

**Review by uranusjr — APPROVED** [review]

**Review by zkan — APPROVED** [review]

**Comment by josh-fell** [comment]
Adding `catchup=False` to DAGs based on [this other PR comment thread](https://github.com/apache/airflow/pull/19237#discussion_r737733552).

**Review by o-nikolas — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
